mint-auditor/src/db/models/audited_mint.rs

```diff
@@ -0,0 +1,696 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use crate::{
+    db::{schema::audited_mints, transaction, Conn, Counters, GnosisSafeDeposit, MintTx},
+    gnosis::{AuditedSafeConfig, GnosisSafeConfig},
+    Error,
+};
+use diesel::prelude::*;
+use serde::{Deserialize, Serialize};
+
+/// Diesel model for the `audited_mints` table.
+/// This stores audit data linking MintTxs with matching GnosisSafeDeposits.
+#[derive(Clone, Debug, Deserialize, Eq, Hash, Insertable, PartialEq, Queryable, Serialize)]
+pub struct AuditedMint {
+    /// Id (required to keep Diesel happy).
+    pub id: Option<i32>,
+
+    /// Id pointing to the MintTx table.
+    pub mint_tx_id: i32,
+
+    /// Id pointing to the GnosisSafeDeposit table.
+    pub gnosis_safe_deposit_id: i32,
+}
+
+impl AuditedMint {
+    /// Attempt to find a matching [MintTx] for a given [GnosisSafeDeposit], and
+    /// if successful return the [MintTx] and record the match in the
+    /// database. Note that each [MintTx] can be matched to at most one
+    /// [GnosisSafeDeposit], so calling this repeatedly over the same
+    /// deposit will fail.
+    pub fn attempt_match_deposit_with_mint(
+        deposit: &GnosisSafeDeposit,
+        config: &AuditedSafeConfig,
+        conn: &Conn,
+    ) -> Result<MintTx, Error> {
+        // We only operate on objects that were saved to the database.
+        let deposit_id = deposit.id().ok_or(Error::ObjectNotSaved)?;
+
+        // The deposit safe needs to match the audited safe configuration.
+        // This shouldn't happen and indicates misuse of this function.
+        if deposit.safe_addr() != &config.safe_addr {
+            return Err(Error::Other(format!(
+                "Gnosis safe deposit addr {} does not match audited safe addr {}",
+                deposit.safe_addr(),
+                config.safe_addr
+            )));
+        }
+
+        let result = transaction(conn, |conn| {
+            // Currently we only support 1:1 mapping between deposits and mints, so ensure
```

Comment:
No plan, but since humans are involved in the minting process in the end can issue whatever transactions they want, we might end up with some odd cases. I am hoping that won't happen since we will provide tooling that should prevent it, but if it does (for example if someone deposits 10 to the safe and then proceeds to mint 9 for whatever reason, and then doing another mint of 1) we might have to add support for that.